src/sha256.rs

```diff
@@ -133,6 +134,62 @@ impl HashTrait for Hash {
     }
 }
 
+/// Output of the SHA256 hash function
+#[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Hash)]
+pub struct Midstate([u8; 32]);
```

Comment:
I think we should make the internal representation public. It's a bit more footgunny but would make the taproot midstates much simpler to implement.

Reply:
How is that different from `from_inner`?

Reply:
`from_inner` is a trait method and is available in generic code

Reply:
Should we then not do that for all Hash types? In fact that would be helpful in some situations where we are using "the one hash" up to "the n hash" for special purposes. (F.e. asset ID and reissuance token calculation.)

Reply:
> `from_inner` is a trait method and is available in generic code

That's also not true. The `from_inner` method on `Midstate` is not a trait method.

Reply:
To conclude, I think we can publish the internal representation until `const fn` is a stable thing, then we can break and have people use `from_inner` instead.